### DISCUSSION TITLE
Combine EOmaps with streamlit?

### DISCUSSION THREAD

**Discussion by past0723** [raphaelquast/EOmaps#126] [Ideas]
Hi Raphael, I just suddenly came up with an idea. What if you or somebody (perhaps me), try to combine EOmaps with Streamlit?

I guess one benefit will be that more users with little or no Python knowledge can be able to use your EOmaps tools, at least locally. 

I understand that a Streamlit APP can be quite big in size. But if just running them locally, then it may not be a big problem? Once a Streamlit App is built by an experienced Python user in a group, he/she can pass it to other users, who will just use the Streamlit APP to carry on some routine analysis?

I am just thinking, not sure about feasibility, challenges, whether it is worth the effort and time.

**Comment by raphaelquast** [comment]
Hey, sorry for taking so long to respond!

I don't have any prior experience with streamlit but I can try to comment a bit on this...
As I see it streamlit seems to focus on web-apps and therefore I would see it more with packages that focus on wrapping javascript libraries like `plotly` `holoviews` etc.. 

If you want a gui wrapped around EOmaps for data-analysis, using `Qt5`, `wxpython` or `tkinter` is probably a more straight-forward way to go to build yourself a custom interface since `matplotlib` figures can easily be embedded in those frameworks.
(have a look at the [companion-widget](https://eomaps.readthedocs.io/en/latest/api.html#companion-widget) for an example how to interact with a `Qt5` widget)

... concerning installing...  
- since `EOmaps` is on conda-forge, installing is as simple as running `mamba install eomaps` in a fresh environment, so that shouldn't actually be much of a problem... 
- ... if you want to have something pre-configured, you could build yourself a docker-container?

**Reply by past0723** [reply]
OK. I thought about streamlit, because I am more familiar with it. There is an example: https://streamlit.geemap.org/

I haven't tried your widget yet, I guess that is why I raised this issue. I once tried to build something with `Qt5`, but I failed. I will have a try later.
